src/MatrixTwitter.js

```diff
@@ -0,0 +1,987 @@
+    if (this.hashtag_intervalID) {
+        clearInterval(this.hashtag_intervalID);
+        this.hashtag_intervalID = null;
+    }
+}
+
+
+/**
+ * MatrixTwitter.prototype.add_hashtag_feed - Add a hashtag to be processed.
+ *
+ * @param  {string} hashtag The hasgtag to add (without the #).
+ * @param  {external:Entry} entry The entry for a room in the RoomBridgeStore.
+ */
+MatrixTwitter.prototype.add_hashtag_feed = function(hashtag,entry){
+    var obj = {
+        "hashtag": hashtag,
+        "entry": entry
+    };
+    this.hashtag_list.push(obj);
+    this.hashtag_queue.push(obj);
+    log.info('Twitter',"Added Hashtag Feed: %s",hashtag);
+}
+
+
+/**
+ * MatrixTwitter.prototype.remove_hashtag_feed - Remove a hashtag so that it
+ * will no longer be processed.
+ *
+ * @param  {string} hashtag The hasgtag to remove (without the #).
+ */
+MatrixTwitter.prototype.remove_hashtag_feed = function(hashtag){
+  const htfind = (feed) => { feed.hashtag == hashtag };
+  this.hashtag_list  = this.hashtag_list.splice( this.hashtag_list.findIndex(tlfind) ,1);
+  this.hashtag_queue = this.hashtag_queue.splice(this.hashtag_queue.findIndex(tlfind),1);
+}
+
+MatrixTwitter.prototype._process_hashtag_feed = function(){
+  if (this.hashtag_queue.length < 1) {
+      return;
+  }
+
+  var feed = this.hashtag_queue.shift();
+  var req = {
+    q: "%23"+feed.hashtag,
+    result_type: 'recent'
+  };
+  var since = feed.entry.remote.get("twitter_since");
+  if (since != undefined) {
+      req.since_id = since;
+  }
+
+  this.app_twitter.get('search/tweets', req, (error, results, response) => {
+      if(error){
+        log.error("Twitter","_process_timeline: GET /statuses/user_timeline returned: %s", error);
+        return;
+      }
+      if(results.statuses.length > 0){
+        feed.entry.remote.set("twitter_since", results.search_metadata.max_id_str);
+        this._bridge.getRoomStore().upsertEntry(feed.entry);
+      }
+
+      results.statuses.reverse().forEach((item) => {
+          this.process_tweet(feed.entry.matrix.roomId, item, 0);
+      });
+  });
+  this.hashtag_queue.push(feed);
+}
+
+/*
+  User Stream functions
+*/
+
+
+/**
+ * MatrixTwitter.prototype.attach_user_stream - Start reading live updates from
+ * a Twitter User Stream.
+ *
+ * @param  {string} user The user's matrix ID.
+ * @return {Promise}   A Promise that will resolve with the operation completes.
+ */
+MatrixTwitter.prototype.attach_user_stream = function(user){
+  if(this.user_streams.hasOwnProperty(user)){
+    log.warn("Twitter","Not attaching stream since we already have one connected!");
+  }
+  return this._get_twitter_client(user).then((c) => {
+    var stream = c.stream('user', {with: "followings"});
+    log.info("Twitter","Attached stream for " + user);
+    stream.on('data',  (data) => {
+      if(data.hasOwnProperty("direct_message")){
+        this._process_incoming_dm(data.direct_message);
+      }
+      else if (data.hasOwnProperty("warning")){
```

Comment:
When you're accessing known properties like this, just do `data.warning` rather than `hasOwnProperty`.